apps/examples/testcase/le_tc/compression/tc_compress_read.c

```diff
@@ -22,12 +22,82 @@
 #include <tinyara/config.h>
 #include <stdio.h>
 #include <stdlib.h>
+#include <errno.h>
 #include <fcntl.h>
 #include <sys/ioctl.h>
+#include <sys/types.h>
 #include <tc_common.h>
 #include <tinyara/os_api_test_drv.h>
+#include <tinyara/compression.h>
 #include "tc_internal.h"
 
+#define INPUT_OUTPUT_SIZE 10000
+
+static struct compress_header *comp_tc_data;
+static struct compress_header *decomp_tc_data;
+
+static void uninitialize_tc_data(void)
+{
+	/* No need to free decomp_tc_data->input_buffer because 
+	 * it is same as comp_tc_data->output_buffer */
+	free(comp_tc_data->input_buffer);
+	free(comp_tc_data->output_buffer);
+	free(decomp_tc_data->output_buffer);
+	free(comp_tc_data);
+	free(decomp_tc_data);
+}
+
+static int initialize_tc_data(void)
+{
+	comp_tc_data = (struct compress_header *)malloc(sizeof(struct compress_header));
+	if (comp_tc_data == NULL) {
+		return ERROR;
+	}
+		
+	comp_tc_data->input_buffer = (unsigned char *)malloc(INPUT_OUTPUT_SIZE);
+	if (comp_tc_data->input_buffer == NULL) {
+		goto exit_with_comp_input;
+	}
+
+	comp_tc_data->input_size = INPUT_OUTPUT_SIZE;
+	
+	for(int i = 0; i < comp_tc_data->input_size; i++) {
+		comp_tc_data->input_buffer[i] = 'a' + (rand() % 26);
+	}
+	
+	comp_tc_data->output_size = INPUT_OUTPUT_SIZE;
+	comp_tc_data->output_buffer = (unsigned char *)malloc(INPUT_OUTPUT_SIZE);
```

Comment:
When we compress and decompress some data, size of input and output buffer is always the same?
When we compress 40 characters, we can allocate 40 char size of output buffer, 100 char output for 100 char input? Is this right?

Reply:
yes, because compressed output size can be input size max but never more than that. But in most cases it will be less than input size . So its better if we allocate same size to both .